math/mathcore/inc/TRandomGen.h

```diff
@@ -52,8 +52,8 @@ class TRandomGen : public TRandom {
    
    TRandomGen(ULong_t seed=1) {
       fEngine.SetSeed(seed);
-      SetName(TString::Format("Random_%s",fEngine.Name().c_str() ) );
-      SetTitle(TString::Format("Random number generator: %s",fEngine.Name().c_str() ));
+      SetName(TString::Format("Random_%s", std::string_view(fEngine.Name()).data()));
```

Comment:
Should we `#include <ROOT/RStringView.hxx>`?

Reply:
In principle - theoretically - `fEngine.Name()` might return a `string_view` into `"NAME followed by garbage"` (restricting the view to the `"NAME"` part). `data()` will expose the garbage as part of the name, too.

Even though less performance, I would thus opt for `std::string(fEngine.Name()).c_str()`.